include/alpaka/mem/buf/BufCpu.hpp

```diff
@@ -82,23 +82,14 @@ namespace alpaka
                 unpin(*this);
 #endif
                 // NOTE: m_pMem is allowed to be a nullptr here.
-                alpaka::free(*this, m_pMem);
-            }
-
-        private:
-            //! \return The number of elements to allocate.
-            template<typename TExtent>
-            ALPAKA_FN_HOST static auto computeElementCount(TExtent const& extent) -> TIdx
-            {
-                auto const extentElementCount = extent::getExtentProduct(extent);
-
-                return extentElementCount;
+                m_fDeleter(m_pMem);
             }
 
         public:
             DevCpu const m_dev;
             Vec<TDim, TIdx> const m_extentElements;
             TElem* const m_pMem;
+            std::function<void(TElem*)> m_fDeleter;
```

Comment:
is `m_deleter` a better choice of name ?

Reply:
Yes please. The `f` reminds me too much of Hungarian notation.